### DISCUSSION TITLE
check for rpmlib version

### DISCUSSION THREAD

**Discussion by oliverkurth** [rpm-software-management/rpm#2841] [Q&A] [answered]
Is there a way to check at compile time for the rpmlib version?

In this particular case I want to use the enum value `RPMTRANS_FLAG_NODB` of `enum rpmtransFlags_e` in `include/rpm/rpmts.h`, but stay compatible with older versions that do not have this implemented. Since it's an enum, I cannot use `#ifdef`. So alternatively I thought I can use a macro that lets me check the version of rpmlib.

**Comment by pmatilai** [comment, marked as answer]
You can test for symbol existence without it being a macro too, details depend on the buildsystem of your project. 
The rpm (not library) version is exposed via RPMVERSION (and rpmEVR) variable when you include rpm/rpmlib.h, but version based checks are best avoided.

**Comment by oliverkurth** [comment]
Thanks for replying.

I have this code:
```
...
                    } else if (strcmp(token, "nodb") == 0){
                        pConf->rpmTransFlags |= RPMTRANS_FLAG_NODB;
                    } else if (strcmp(token, "noconfigs") == 0){
                        pConf->rpmTransFlags |= RPMTRANS_FLAG_NOCONFIGS;
                    } else {
                        pr_err("unknown tsflag '%s'\n", token);
...
```
so for `RPMTRANS_FLAG_NODB` I want to build this conditionally, depending on whether `RPMTRANS_FLAG_NODB` is defined or not. As I said, I cannot use `#ifdef RPMTRANS_FLAG_NODB` because it's an enum value, not a macro. What do you recommend to do here?

FWIW, the project uses `cmake`. It's https://github.com/vmware/tdnf . Of course I could create some setting like `HAVE_RPMTRANS_FLAG_NODB` and set that as an argument to `cmake`, but I'd prefer something automatic.

**Reply by pmatilai** [reply]
FWIW, this isn't an rpm issue at all. It's like any old feature test, you test for the symbol in cmake, and define something you *can* test with ifdef. There are multiple ways to do that.

**Reply by oliverkurth** [reply]
I got that. It would be good if `rpmlib` had a macro that can just be tested, to save some work, and not having to write a hack to test for it. If you (or anyone) knows of any similar code that does that, it would be great if you can share.
